module/Activity/src/Activity/Service/Activity.php

```diff
@@ -110,12 +111,30 @@ public function createActivity(array $params)
         // Find the creator
         $user = $this->getServiceManager()->get('user_role');
         if ($user === 'guest') {
-            throw new \InvalidArgumentException('Guests can not create activities');
+            throw new NotAllowedException('Guests can not create activities');
         }
-        $params['creator'] = $user;
-
         $activity = new ActivityModel();
-        $activity->create($params);
+        $activity->setBeginTime(new \DateTime($params['beginTime']));
+        $activity->setEndTime(new \DateTime($params['endTime']));
+        $activity->setName($params['name']);
+        $activity->setNameEn($params['name_en']);
+        $activity->setLocation($params['location']);
+        $activity->setLocationEn($params['location_en']);
+
+        if (!$params['costs_unknown']) {
+            $activity->setCosts($params['costs']);
+            $activity->setCostsEn($params['costs_en']);
+        }
```

Comment:
If a form only defines what a user sees, then we might as well not use forms, since the form is being rendered manually anyways (https://github.com/stefanmajoor/gewisweb/blob/03a585790c2e17bcf8df33da60a396aedfe2b709/module/Activity/view/activity/activity/create.phtml).